policy-controller/k8s/index/src/index.rs

```diff
@@ -1070,13 +1082,53 @@ impl Pod {
             );
         };
 
+        // No Server is configured, so requests on the Pod's probe paths are
+        // authorized.
+        let http_routes = {
+            let probe_paths = probes.get(&port).into_iter().flatten().map(|p| p.as_str());
+            Self::http_probe_routes(probe_paths, &config.probe_networks)
+        };
+
         InboundServer {
             reference: ServerRef::Default(policy.to_string()),
             protocol,
             authorizations,
-            http_routes: HashMap::default(),
+            http_routes,
         }
     }
+
+    fn http_probe_routes<'p>(
+        paths: impl Iterator<Item = &'p str>,
+        networks: &[IpNet],
+    ) -> HashMap<String, InboundHttpRoute> {
+        paths
+            .map(|path| {
+                let authz = Some((
+                    AuthorizationRef::Default("probe".to_string()),
+                    ClientAuthorization {
+                        networks: networks.iter().copied().map(Into::into).collect(),
+                        authentication: ClientAuthentication::Unauthenticated,
+                    },
+                ));
+
+                let route = InboundHttpRoute {
+                    hostnames: vec![],
+                    rules: vec![InboundHttpRouteRule {
+                        matches: vec![HttpRouteMatch {
+                            path: Some(PathMatch::Exact(path.to_string())),
+                            headers: vec![],
+                            query_params: vec![],
+                            method: Some(Method::GET),
+                        }],
+                        filters: vec![],
+                    }],
+                    authorizations: authz.into_iter().collect(),
+                };
+
+                (path.to_string(), route)
```

Comment:
Hmmm I think it's wrong to return `path` here. We need to return a "route name" and, until now, we've assumed routes are all `HTTPRoute` resources. Now, we're going to have to differentiate between defaults and resources by key type, and this will extend into how routes are serialized into protobuf.

Reply:
Before you fix this you should pull in changes from `main` that include #9009